### PR TITLE
feat: connector DB schema

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

Create connector DB schema.

Notice that some info are stored in connector's code, see [LOG-1267](https://linear.app/silverhand/issue/LOG-1267/config-file-schema)

## Linear Issue Reference

[LOG-1202](https://linear.app/silverhand/issue/LOG-1202/schema)

## Reviewer

@gao-sun @simeng-li cc: @darcyYe

### CODE REVIEW

File: packages/schemas/tables/connectors.sql [hunk 1]
```diff
@@ -0,0 +1,12 @@
+create type connector_type as enum ('SMS', 'Email', 'Social');
+
+create table connectors (
+  id varchar(128) not null,
+  identifier varchar(256) not null unique,
+  enable boolean not null default TRUE,
```
Comment (gao-sun) [inline packages/schemas/tables/connectors.sql:6]: discussion: should we add prefix `is`, `has` for boolean fields? @logto-io/eng

Comment (gao-sun) [inline packages/schemas/tables/connectors.sql:6]: `enabled`

File: packages/schemas/tables/connectors.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  type connector_type not null,
+  config jsonb /* @use ConnectorConfig */ not null default '{}'::jsonb,
+  data jsonb /* @use ConnectorData */ not null default '{}'::jsonb,
```
Comment (gao-sun) [inline packages/schemas/tables/connectors.sql:9]: looks like these two fields will be updated at the same time. maybe `config` is enough?

**Review by gao-sun — CHANGES_REQUESTED** [review]

File: packages/schemas/tables/connectors.sql (hunk 2 above)
Comment (wangsijie) [reply packages/schemas/tables/connectors.sql:9]: `config`: updated at admin console by management API (developer).
`data`: updated at connector runtime by connector itself, for instance: update template review status.

But merge them into a single `config` is also acceptable, any idea? @simeng-li

File: packages/schemas/tables/connectors.sql (hunk 1 above)
Comment (darcyYe) [reply packages/schemas/tables/connectors.sql:6]: I prefer boolean fields with prefixes.

File: packages/schemas/tables/connectors.sql [hunk 3]
```diff
@@ -0,0 +1,12 @@
+create type connector_type as enum ('SMS', 'Email', 'Social');
+
+create table connectors (
+  id varchar(128) not null,
+  identifier varchar(256) not null unique,
```
[the remark this answers is not in the document]
Comment (darcyYe) [reply packages/schemas/tables/connectors.sql:5]: I think two 'id' fields are required. Developers might list all provided connectors and choose among them. We may expose one 'id' field and they can specify which connector to use with the exposed 'id's. And the actual 'id's backend was used to distinguish different connectors should be invisible to developers for security concerns.

File: packages/schemas/tables/connectors.sql (hunk 2 above)
Comment (gao-sun) [reply packages/schemas/tables/connectors.sql:9]: things like template review status should be a real-time fetching stuff?

File: packages/schemas/src/db-entries/connector.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  enable: boolean;
+  type: ConnectorType;
+  config: ConnectorConfig;
+  data: ConnectorData;
+  createdAt: number;
+};
+
+const guard: Guard<ConnectorDBEntry> = z.object({
+  id: z.string(),
+  identifier: z.string(),
```
Comment (gao-sun) [inline packages/schemas/src/db-entries/connector.ts:37]: we need re-gen this file when schema changed

File: packages/schemas/tables/connectors.sql (hunk 2 above)
Comment (wangsijie) [reply packages/schemas/tables/connectors.sql]: make sense, remove `data` for now.

File: packages/schemas/src/foundations/jsonb-types.ts [hunk 5]
```diff
@@ -31,3 +31,7 @@ export const userLogPayloadGuard = z.object({
 });
 
 export type UserLogPayload = z.infer<typeof userLogPayloadGuard>;
+
+export const connectorConfigGuard = z.object({});
```
Comment (simeng-li) [inline packages/schemas/src/foundations/jsonb-types.ts:35]: Add a TODO comment?

**Review by simeng-li — APPROVED** [review]
LGTM

File: packages/schemas/tables/connectors.sql (hunk 2 above)
Comment (simeng-li) [reply packages/schemas/tables/connectors.sql]: merge +1

File: packages/schemas/tables/connectors.sql (hunk 1 above)
Comment (gao-sun) [reply packages/schemas/tables/connectors.sql]: any ideas? @simeng-li @IceHe

File: packages/schemas/src/foundations/jsonb-types.ts (hunk 5 above)
Comment (wangsijie) [reply packages/schemas/src/foundations/jsonb-types.ts:36]: `// TODO ` (with trailing space) works too.

File: packages/schemas/tables/connectors.sql (hunk 1 above)
Comment (IceHe) [reply packages/schemas/tables/connectors.sql]: prefer `enabled`

**Review by darcyYe — APPROVED** [review]
lgtm

File: packages/schemas/src/foundations/jsonb-types.ts (hunk 5 above)
Comment (gao-sun) [reply packages/schemas/src/foundations/jsonb-types.ts:36]: didnt see the lint complaint in GitHub, would you like to add an issue to see what happened? should be able to see the warning here.

**Review by gao-sun — APPROVED** [review]

File: packages/schemas/src/foundations/jsonb-types.ts (hunk 5 above)
Comment (wangsijie) [reply packages/schemas/src/foundations/jsonb-types.ts:36]: Change to `// TODO:` still not working, created LOG-1289

**Pull request merged**